apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -661,10 +661,13 @@ public function createShare(
 
 		// Handle mail send
 		if (is_null($sendMail)) {
-			// Define a default behavior when sendMail is not provided
-			// For email shares with a valid recipient, the default is to send the mail
-			// For all other share types, the default is to not send the mail
-			$allowSendMail = ($shareType === IShare::TYPE_EMAIL && $shareWith !== null && $shareWith !== '');
+			$allowSendMail = $this->config->getSystemValueBool('sharing.enable_share_mail', true);
```

Comment:
```suggestion
			$allowSendMail = $this->config->getSystemValueBool('sharing.enable_share_mail', true) || ($shareType === IShare::TYPE_EMAIL && $shareWith !== null && $shareWith !== '');
```
The entire logic could be written a bit nicer like this, since it deduplicates checks.